### PR TITLE
feat(components): SSR support for toast

### PR DESCRIPTION
**Pull request by MichaelParadis**
## Motivations

During our SSR support updates we missed adding support for toasts. This PR fixes that by preventing references to `document` if it is undefined

### Added

- Added SSR support for toasts

### Changed

- 

### Deprecated

- 

### Removed

- 

### Fixed

- 

### Security

- 

## Testing
Verify Toasts work in the referenced PRs

Changes can be
[tested via Pre-release](https://github.com/GetJobber/atlantis/actions/workflows/trigger-qa-build.yml)

---

[In Atlantis we use Github's built in pull request reviews](https://help.github.com/en/articles/about-pull-request-reviews).

Random photo of Atlantis

### CODE REVIEW

File: packages/components/src/Toast/Toast.test.tsx [hunk 1]
```diff
@@ -96,15 +103,15 @@ it("stops and starts the timer when the item is hover toggled", async () => {
 
   fireEvent.mouseEnter(getByText("Bland Toast"));
 
-  await act(() => {
+  act(() => {
```
Comment (MichaelParadis) [inline packages/components/src/Toast/Toast.test.tsx:106]: These awaits were not actually needed.

**Comment by darryltec** [comment]
Code looks good. Mind testing it in remix where you render it in the server. I wonder if that shows up on the client after mount or not at all until you fire it in the client again

**Comment by MichaelParadis** [comment]
@darryltec I have a referenced PR. I had to fire it again on the client since I wasn't able to find a way to modify the document while hydrating (I could have easily missed something)

**Comment by scotttjob** [comment]
> should also export this in here `packages/components/src/index.tsx` so you can import from `@jobber/components` directlly

Only if this has been tested elsewhere to make sure it doesn't break existing implementations. That's why it was left out to begin with.

**Comment by MichaelParadis** [comment]
@scotttjob I assume it was left out initially because toast would break on SSR usages. However since this fixes that it should be good. I will do another pre-release and verify

**Comment by scotttjob** [comment]
> @scotttjob I assume it was left out initially because toast would break on SSR usages. However since this fixes that it should be good. I will do another pre-release and verify

That's correct. However when I fixed the SSR issues in a last minute effort when I realized they were broken, it caused issues in other repos. So just make sure it's gravy everywhere and we should be good to go.

**Comment by MichaelParadis** [comment]
>That's correct. However when I fixed the SSR issues in a last minute effort when I realized they were broken, it caused issues in other repos. So just make sure it's gravy everywhere and we should be good to go.

I was able to verify that this worked in the other places 😉

**Comment by MichaelParadis** [comment]
@darryltec updated in 4b12bdf465f6d1e8abc0ef645c19dc71b2b9ad27

**Review by darryltec — APPROVED** [review]
:shipit:

**Pull request merged**